spring-restdocs/src/main/java/org/springframework/restdocs/hypermedia/Field.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2014-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.hypermedia;
```

Comment:
I agree.
For inspiration I took a closer look at how the [HAL specification](http://stateless.co/hal_specification.html) calls these fields. 
The HAL model is making the distinction between links, embedded resources, and plain old state (more specifically called "your bog standard JSON or XML data").
So I moved the "fields" classes to org.springframework.restdocs.state.